xclim/indices/_agro.py

```diff
@@ -580,6 +584,167 @@ def water_budget(
     return out
 
 
+@declare_units(
+    pr="[precipitation]",
+)
+def standardized_precipitation_index(
+    pr: xarray.DataArray,
+    pr_cal: xarray.DataArray,
+    window: int = 1,
+    dist: str = "gamma",
+) -> xarray.DataArray:
+    r"""Standardized Precipitation Index (SPI).
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+      Daily precipitation.
+    pr_cal : xarray.DataArray
+      Daily precipitation used for calibration.
+    window : int
+      Averaging window length (in months).
+    dist : str
+      Name of the univariate distribution, such as `beta`, `expon`, `genextreme`, `gamma`, `gumbel_r`, `lognorm`, `norm`
+      (see scipy.stats).
+
+    Returns
+    -------
+    xarray.DataArray,
+      Standardized Precipitation Index.
+
+
+    Notes
+    -----
+    The length `N` of the N-month SPI is determined by choosing the `window = N`.
+
+
+    Example
+    -------
+    Computing SPI-3 using a gamma distribution for the fit
+
+    .. code-block:: python
+
+          import xclim.indices as xci
+          import xarray as xr
+
+          ds = xr.open_dataset(filename)
+          pr = ds.pr
+          pr_cal = pr.sel(time=slice(calibration_start_date, calibration_end_date)
+          window = 3
+          dist = "gamma"
+          spi_3 = xci._agro.standardized_precipitation_index(pr, pr_cal, window, dist)
+
+
+    References
+    ----------
+    .. [1] McKee, Thomas B., Nolan J. Doesken, and John Kleist. "The relationship of drought frequency and duration to time scales." In Proceedings of the 8th Conference on Applied Climatology, vol. 17, no. 22, pp. 179-183. 1993.
+    """
+    supported_dists = ["gamma"]
+    if dist not in supported_dists:
+        raise NotImplementedError(f"Distribution `{dist}` not supported.")
+
+    freq = "MS"
+
+    # Resampling and rolling precipitations
+    pr = pr.resample(time=freq).mean()
+    if window > 1:
+        attrs = pr.attrs.copy()
+        pr = pr.rolling(time=window).mean(skipna=False)
+        pr.attrs.update(attrs)
+
+    # Obtain fitting params and expand along time dimension
+    params = fit_group(pr_cal, dist, freq, window)
+    grouped_time = list(pr.time.groupby("time.month"))
+    params = xarray.concat(
+        [
+            params.isel(month=i)
+            .drop_vars("month")
+            .assign_coords(time=grouped_time[i][1][0].values)
+            .expand_dims("time")
+            .reindex_like(grouped_time[i][1], method="ffill")
+            for i in range(0, params.sizes["month"])
+        ],
+        dim="time",
+    )
```

Comment:
```suggestion
    params = params.rename(month='time').reindex(time=pr.time.dt.month)
```
I believe this will have the same effect, with the bonus of not relying on an explicit loop.